packages/cron/src/jobs/pins-backup.js

```diff
@@ -0,0 +1,301 @@
+            Key: key,
+            Body: bak.content,
+            Metadata: { structure: 'Complete' }
+          }
+        })
+        await upload.done()
+      }
+    }
+    this.log(`✅ Finished uploading ${bak.sourceCid}`)
+    return url
+  }
+
+  /**
+   * @param {import('@nftstorage/ipfs-cluster').Cluster} ipfs
+   * @param {Array<import('@nftstorage/ipfs-cluster/src/interface.js').ClusterInfo>} peersList
+   * @param {Object} [options]
+   * @param {number} [options.maxDagSize] Skip DAGs that are bigger than this.
+   */
+  exportCar (ipfs, peersList, options = {}) {
+    /**
+     * @param {AsyncIterable<import('../types/pins-backup').BackupCandidate>} source
+     * @returns {AsyncIterableIterator<import('../types/pins-backup').BackupContent>}
+     */
+    const _this = this
+    return async function * (source) {
+      for await (const pin of source) {
+        const peer = peersList.find((peer) => peer.ipfs.id === pin.peer)
+        if (peer) {
+          const content = await _this.ipfsDagExport(pin.sourceCid, peer.ipfs.addresses[0], options)
+          yield { ...pin, content }
+        } else {
+          _this.log(`🚨 ${pin.sourceCid} has not been found on cluster`)
+          throw (new Error('Not on cluster'))
+        }
+      }
+    }
+  }
+
+  /**
+   * Export a CAR for the passed CID.
+   *
+   * @param {import('multiformats').CID} cid
+   * @param {string} peer The IPFS peer that the CID is available on
+   * @param {Object} [options]
+   */
+  async ipfsDagExport (cid, peer, options) {
+    const { writer, out } = await CarWriter.create([cid])
+    const readableStream = Readable.from(out)
+
+    // Do not need to await this.
+    ;(async () => {
+      const abortController = new TimeoutController(20_000)
+      let bytesReceived = 0
+
+      let reportInterval
+      const libp2p = await getLibp2p()
+      try {
+        const dagula = await Dagula.fromNetwork(libp2p, { peer })
+        let bytesTotal
+
+        reportInterval = setInterval(() => {
+          const formattedTotal = bytesTotal ? this.fmt(bytesTotal) : 'unknown'
+          this.log(`ℹ️ CID: ${cid}. Received ${this.fmt(bytesReceived)} of ${formattedTotal} bytes`)
+        }, this.REPORT_INTERVAL)
+
+        this.log(`⏳ Started reading dag ${cid}`)
+
+        for await (const chunk of dagula.get(cid, {
+          signal: abortController.signal
+        })) {
+          abortController.reset()
+          bytesReceived += chunk.bytes.length
+          await writer.put(chunk)
+        }
+
+        this.log(`✅ Finished reading dag ${cid}. In total ${this.fmt(bytesReceived)}`)
+      } catch (err) {
+        this.log(`🚨 CID: ${cid} errored. Size of the dag is greater than ${bytesReceived} Bytes`)
+        readableStream.destroy(new Error('Dagula errored while reading'))
+        throw (err)
+      } finally {
+        if (bytesReceived > this.MAX_UPLOAD_DAG_SIZE) {
+          this.log(`⚠️ CID: ${cid} dag is greater than ${this.fmt(this.MAX_UPLOAD_DAG_SIZE)}`)
```

Comment:
Add a per batch summary:
- failed cids (and their size)
- Successful (bigger than MAX_UPLOAD_DAG_SIZE)

Remove all unnecessary per cids logging, log just errors by default.

Reply:
Done!

Default logging (`DEBUG=backupPins:log` ) is quite succinct now, while the job can be run manually passing a more verbose `DEBUG=backupPins:*` through workflow inputs.

Example of default logging:
``` 
❯ DEBUG=backupPins:log npm test --workspace=packages/cron
```
![Screenshot 2022-10-31 at 16 24 12](https://user-images.githubusercontent.com/3163684/199058522-e6ad8729-246e-4466-9721-78958f1324af.png)

``` 
❯ DEBUG=backupPins:log npm test --workspace=packages/cron
```
![Screenshot 2022-10-31 at 16 26 17](https://user-images.githubusercontent.com/3163684/199058942-1fdae1dd-5ddc-4564-afcd-22f5be054e8c.png)